### PR TITLE
Add VPM support

### PR DESCRIPTION
**Pull request by jellejurre**
I'm quite new to this, but I followed the steps from https://vcc.docs.vrchat.com/guides/convert-unitypackage to add VCC/VPM support to the emulator, and from my testing it works as a VPM package now. I was unsure what to enter for the package.json, but made sure to change the absolute path to a local path (as that wouldn't be the first time that's doxed someone xd).

### CODE REVIEW

**Comment by jellejurre** [comment]
I want to mention I did not add any of the github workflows from https://github.com/vrchat-community/template-package, just moved folders around, created package.json, and added assembly definitions

**Comment by jellejurre** [comment]
Actually, when testing uploading, it seems some parts need extra "If Editor" checks, since now it wont let me upload due to deprecated APIs, will fix later today

**Comment by jellejurre** [comment]
Fixed

**Comment by jellejurre** [comment]
Updated gesture manager support to match the vcc version.

To enable gesture manager right now, you need to commend out the `#define DISABLE_GESTURE_MANAGER` line in GestureManagerAv3MenuEditor.cs, and add the gesture manager assemblies to the editor assembly like so: 
image

**Comment by jellejurre** [comment]
Added detection as well so it automatically does that if GestureManager is found on disk. I do it in a kinda janky way by directly editing the assembly reference, but I tested it and it works

**Comment by jellejurre** [comment]
By the way, once a user uninstalls GestureManager, the package breaks. I wouldn't know how to recover from having a project with imports of a folder that doesn't exist. It is fixed by reinstalling the emulator.

**Comment by lyuma** [comment]
Hey. Sorry I've been sort of subsumed by work the past couple weeks. Just catching up on stuff.

Wow! 🙏 thank you so much for doing all of this.

So one of the things I want to figure out is how to do the connection to GestureManager in a way that avoids maintenance headaches for either package.

I'd prefer avoiding any of the complexity around gesture manager with the GestureManagerBridge. It is supposed to be possible to include defines depending on whether or not a VPM package is in the project, such as:
https://docs.unity3d.com/2019.4/Documentation/Manual/ScriptCompilationAssemblyDefinitionFiles.html#define-symbols

so the simple #ifdef should be good enough. If this isn't working, we might need to come up with a different strategy, such as making a copy of the older gesture manager in the Av3Emulator namespace and removing all external facing parts so it becomes just an internal component to av3 emulator. But I'd rather see if we can reference the package normally.

I saw some inconsistencies between guids that got changed. I might suggest to change the AvatarMask instances to be created in code, since they are all relatively simple and should avoid the complexity of having hardcoded guid and file path references, since this was a big headache during the VPM transition.

I'm a little apprehensive about having my name be in the package name :-p I'm hoping for there to be more community involvement in the future, such as contributions like this one.

Another thing to figure out is if there's perhaps a new namespace that Av3Emulator can be moved to in order to avoid conflicts with the non-VPM version. This is similar to how GestureManager is now wrapped in the BlackStartX namespace.

**Comment by jellejurre** [comment]
I'll try looking into the mask & assembly definition stuff tomorrow if I've got time

As for the namespace and package name, do you have any suggestions?

I feel like generally having the name of the creator as the namespace is quite standard, even with a more community-driven project, because it ensures not getting conflicts

**Comment by lyuma** [comment]
hey you could look at https://github.com/anatawa12/Av3Emulator-VCC
they did the asmdef correctly

I guess we now have two people working on this ;-)

File: package.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+  "name": "lyuma.av3emulator",
```
Comment (anatawa12) [inline package.json:2]: In my opinion, It's much better to use `io.github.lyuma` as prefix of the package. 
lyuma.github.io is domain that usable by github user lyuma (as github pages) and `io.github.<github user>` is used as prefix for who doesn't own any domain in mavem central, another package system that uses reverse domain notation.

Comment (jellejurre) [reply package.json:2]: Im fine with using any prefix, but this still has Lyumas name in the package, so I don't know how they feel about it @lyuma

**Comment by jellejurre** [comment]
I moved the controller & masks to code, although I do not know much about the transform part of the masks. I looked at the way it was currently set up, and it used a whole bunch of transforms which I feel don't really align with what's necessarily on the avatar?

I wasn't sure what to do here, so for now I didn't touch that part of the masks, but if more changes are needed, please do let me know.

**Comment by jellejurre** [comment]
@lyuma sorry to ping, but what is the status on this?
I believe you want me to change the namespace and package name? Would something like just AV3Emulator or community.AV3Emulator work? I'm kinda afraid of clashes there, but that way we can avoid having your name in there

**Comment by jellejurre** [comment]
I believe this should be all the meta files changed back. When I look at the changed files all changes are sensible now

**Comment by lyuma** [comment]
Thanks so much for all the maintenance you've been taking care of past few weeks.

I'm not sure if it's a good idea to have the same GUID in packages and assets. I would have assumed that you want all GUIDs to be different so unitypackage installs don't overwrite packages files in weird ways

**Pull request merged**